src/functionBased/components/TodoContainer.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable comma-dangle */
+/* eslint-disable no-param-reassign */
+import React, { useState, useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import Header from './Header';
+import InputTodo from './InputTodo';
+import TodosList from './TodoLists';
+
+function getInitialTodos() {
+  const storedTodos = localStorage.getItem('todos');
+  const savedTodos = JSON.parse(storedTodos);
+  return savedTodos || [];
+}
+
+const TodoContainer = () => {
+  const [todos, setTodos] = useState(getInitialTodos());
+
+  const handleChange = (id) => {
+    setTodos((prevState) => {
+      prevState.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          };
+        }
+        return todo;
+      });
+    });
+  };
```

Comment:
Consider refactoring the above function to 

> `const handleChange = (id) => {
    setTodos((prevState) =>  prevState.map((todo) => {
      if (todo.id === id) {
        return {
          ...todo,
          completed: !todo.completed,
        };
      }
      return todo;
    }));
  };`
  
This is the reason why the code breaks when the check button is clicked to make an item as completed. Notice how the curly braces wrapping the `prevState.map` was removed.

Reply:
Thank you @clintonjosephs for the detailed reviews and suggestions :muscle: